lib/active_model_serializers/adapter/json_api/relationship.rb

```diff
@@ -34,6 +34,11 @@ def as_json
 
         private
 
+        def ensure_valid!(json)
```

Comment:
Created an issue for jsonapi: https://github.com/json-api/json-api/issues/1092, we'll see if it gets any traction...